Sources/Core/Views/ProfileImageView/ProfileImageView.swift

```diff
@@ -0,0 +1,168 @@
+        Badge(style: properties.badging, size: .init(width: geo.size.width / 3.5, height: geo.size.height / 3.5), badgeTapped: tiggerBadgeTapClosure)
+      }
+    }
+  }
+  
+  private func triggerImageTapClosure() { imageTapped?() }
+  private func tiggerBadgeTapClosure() { badgeTapped?() }
+  
+  private struct Badge: View {
+    @State var style: BadgeStyle
+    @State var size: CGSize
+    var badgeTapped: (() -> Void)
+
+    
+    init?(style: BadgingMode, size: CGSize, badgeTapped: @escaping () -> ()) {
+      self.badgeTapped = badgeTapped
+      
+      switch style {
+      case .none:
+        return nil
+      case .some(let badge):
+        self.style = badge
+        self.size = size
+      }
+    }
+    
+    var body: some View {
+      style.image
+        .foregroundColor(.white)
+        .frame(width: size.width, height: size.height)
+        .background(style.backgroundColor)
+        .clipShape(Circle())
+        .gesture(TapGesture().onEnded({ badgeTapped() }))
+        .overlay(Circle().stroke(
+          (style.borderColor != nil) ? style.borderColor! : Color.white,
+          lineWidth: (style.borderWidth != nil) ? style.borderWidth! : 2)
+        )
+    }
+  }
+  
+  private struct ProfileImage: View {
+    @ObservedObject var properties: ProfileImageProperties
+    var profileTapped: (() -> Void)
+    var optionalPlaceHolder: UIImage?
+    
+    var body: some View {
+      GeometryReader { geo in
+        constructProfileImage(data: properties.urlData, placeHolder: properties.placeHolder)
+          .resizable()
+          .aspectRatio(contentMode: properties.contentMode)
+          .frame(width: geo.size.width, height: geo.size.height)
+          .gesture(TapGesture().onEnded({ profileTapped() }))
+          .background(getBackground())
+          .cornerRadius(properties.cornerRadius)
+          .border(properties.borderColor, width: properties.borderWidth)
+      }
+    }
+    
+    private func constructProfileImage(data: Data?, placeHolder: Image) -> Image {
+      switch data {
+      case .some(let data):
+        if let createdImage =  UIImage(data: data) {
+          return Image(uiImage: createdImage)
+        } else {
+          return placeHolder
+        }
+      case .none:
+        return placeHolder
+      }
+    }
+    
+    private func getBackground() -> AnyView {
+      switch properties.backgroundType {
+      case .plain(let backgroundColor):
+        return AnyView(backgroundColor)
+      case let .linearGradient(gradient):
+        return AnyView(gradient)
+      case let .angularGradient(gradient):
+        return AnyView(gradient)
+      case let .radialGradient(gradient):
+        return AnyView(gradient)
+      }
+    }
+  }
+}
+
+// MARK: - Public Methods
+@available(iOS 13, *)
+public extension ProfileImageView {
+  func set(_ url: URL?) {
+    switch url {
+    case .some(let url):
+      URLSession.shared.dataTask(with: url) { data, response, error in
```

Comment:
I like what @tonikocjan did. Can you check [it](https://github.com/poviolabs/PovioKit/blob/develop/Sources/Core/Utilities/ImageSource/ImageSource.swift#L37) out and see if we can use it here as well?

Reply:
It looks like that requires the app to have `kingfisher` no? If that framework isn't present should we fallback to this implementation then? What are your thoughts?

Reply:
It does not require it. It just provides a default implementation of the `resolve` method when Kingfisher is available. You can provide a default implementation when it's not available.

Reply:
Actually I like this approach. Usually we use Kingfisher on every project where images are involved. So that is the happy path. But if for some reason there is no Kingfisher dependency, we can still provide our custom implementation.

Reply:
I would extend `ImageSource` and add `resolve` method implemented using `URLSession.shared.dataTask(with: url)` like it's done now.

Reply:
Great. @ArlindDushi can we do that?

Reply:
@borut-t  Yes, I like the idea. Will work on that.